### PR TITLE
Example/searchbox - added basic searchbox example

### PR DESCRIPTION
**Pull request by callamwilliams**
Got around to adding a SearchBox example as referenced here

https://github.com/google-map-react/google-map-react/issues/460

Not sure if you want any description for it, as it looks like showing the code is probably easier moving forward?

Currently having one issue when placing the marker, which is the same as this

https://github.com/google-map-react/google-map-react/issues/381

`NaN is an invalid value for the left css style property`

Not sure if you have any ideas?

Let me know if you want to change/add anything

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -20,15 +17,19 @@ import App from './App';
...
-  <Router>
-    <App>
-      <Switch>
-        <Route exact path={process.env.BASE_PATH} component={Main} />
-        {/* New examples here */}
-        <Redirect exact from="*" to={process.env.BASE_PATH} />
-      </Switch>
-    </App>
-  </Router>,
+  <div>
```
Comment (itsmichaeldiego) [inline src/index.js]: This div is not necessary, right?

File: .env.sample [hunk 2]
```diff
@@ -1,2 +1,2 @@
-BASE_PATH = /google-map-react-examples/
+REACT_APP_BASE_PATH = /
```
Comment (itsmichaeldiego) [inline .env.sample]: Why this change? We need this base path to make it work in github.io

**Comment by itsmichaeldiego** [comment]
@callamwilliams Thank you so much for this! Regarding your problem, it seems that you're passing an invalid value, when is this happening exactly?

File: src/index.js (hunk 1 above)
Comment (callamwilliams) [reply src/index.js]: Yep, left over because I wanted to log the BASE_PATH - shall remove

File: .env.sample (hunk 2 above)
Comment (callamwilliams) [reply .env.sample]: To make it work locally - I had an issue where CRA couldn't see the var unless it was prefixed with `REACT_APP_ `

The change of the path was to have the homepage fire up once you run `yarn start`

**Comment by callamwilliams** [comment]
I think.... it's because the data from places.json is a set of number coordinates, yet the results returned from the searchbox request are functions. Might need some extra logic to convert these.

**Comment by callamwilliams** [comment]
Yep - just needed to call them as a function rather than an object

File: .env.sample (hunk 2 above)
Comment (itsmichaeldiego) [reply .env.sample]: @callamwilliams Sounds good, but we might need to keep it as `/google-map-react-examples` in order to make it work live here:
- https://google-map-react.github.io/google-map-react-examples/

Comment (callamwilliams) [reply .env.sample]: OK, updated - removed `.env.sample` to keep it clean

Comment (itsmichaeldiego) [reply .env.sample]: @callamwilliams Looks great, did you check in your fork that the `gh-pages` work? The info is in the readme. I think you might need to set up the environment to prod in the `deploy` method of `package.json` so it knows which `.env` file to take. Does this makes sense?

Comment (callamwilliams) [reply .env.sample]: Had to adjust the paths based on environments a little, but working correctly now - GH pages sets process.env to production automatically when pushed which is nice.

screenshot 2018-11-03 at 16 31 26

Comment (itsmichaeldiego) [reply .env.sample]: Thanks @callamwilliams, great work really, I just approved, waiting for circle-ci to do the checks in order to merge.

**Comment by itsmichaeldiego** [comment]
@callamwilliams I've no idea why circle-ci is not allowing me to merge 🤔

**Comment by callamwilliams** [comment]
I'll have a look

**Comment by callamwilliams** [comment]
I've merged on my end - no change

Is it [this?](https://circleci.com/docs/2.0/workflows-waiting-status/)

**Comment by callamwilliams** [comment]
Any luck?

**Comment by itsmichaeldiego** [comment]
@callamwilliams Not yet, but in that post you shared it says _Note: "continuous-integration/travis-ci" is the Status API event which is deprecated_ and that is exactly the api I am using and its not working here. I will try to fix that

**Comment by itsmichaeldiego** [comment]
@callamwilliams I will merge it and fix CI later!

**Comment by itsmichaeldiego** [comment]
@callamwilliams Working on my end! Thank you!

A couple of comments:

I think there are a couple of things we could improve style-wise, like increasing the size of the search input, and reducing the height of the map to `100vh - headerHeight`, otherwise looks totally fine.

There was pointed out [here](https://github.com/google-map-react/google-map-react/issues/460#issuecomment-435697889) that the search input could be inside the map, maybe that is something we can do?

**Comment by itsmichaeldiego** [comment]
@callamwilliams I like your thoughts, I think all your points all valid, and I agree totally that I prefer those changes to be addressed in other PRS.

- I think we could just increase the height and width of the search input, and then set-up the height of google maps so we don't have to scroll down to see the rest of the map, otherwise as I mentioned, we could just put the search input inside the map as @tim-soft did  [here](https://github.com/google-map-react/google-map-react/issues/460#issuecomment-435697889)!

- Regarding some kind of navigation away from each example, I think ideally I would create a navigation bar at the left, so we can navigate easily between them.

- Regarding the autocomplete search, I think that would make sense, yes. 

Thanks again for your help and your thoughts, I appreciate all the time you spent on this, we really need help on this library to make it better. if you would like we could exchange emails and if interested you could be a core-member.

**Comment by itsmichaeldiego** [comment]
@callamwilliams Just sent me an email to itsmichaeldiego@gmail.com, that would work
